cli/cmd/registry_add_gcr_test.go

```diff
@@ -0,0 +1,73 @@
+package cmd
+
+import (
+	"os"
+	"testing"
+	"text/tabwriter"
+)
+
+func TestRegistryAddGCR(t *testing.T) {
+	tests := []struct {
+		name          string
+		args          []string
+		wantErr       bool
+		wantErrString string
+	}{
+		{
+			name:          "endpoint required",
+			args:          []string{"registry", "add", "gcr"},
+			wantErr:       true,
+			wantErrString: "endpoint must be specified, serviceaccountkey or serviceaccountkey-stdin must be specified",
+		},
+		{
+			name:          "service account key required",
+			args:          []string{"registry", "add", "gcr", "--endpoint", "gcr.io"},
+			wantErr:       true,
+			wantErrString: "serviceaccountkey or serviceaccountkey-stdin must be specified",
+		},
+		{
+			name:          "invalid service account key",
+			args:          []string{"registry", "add", "gcr", "--endpoint", "gcr.io", "--serviceaccountkey", "./testdata/invalid-gcr-service-account-key.json"},
```

Comment:
This test should not use relative file paths.  We can use`embed.FS` from the `https://pkg.go.dev/embed` package

Code should be refactored to validate regardless of whether JSON is coming from a file or from STDIN.